Remove calling convention modifiers from cpuid sig

am11 opened the pull request:
Failure on linux-musl-x86:
> /runtime/src/coreclr/vm/cgensys.h:107:26: error: stdcall calling convention is not supported on builtin function [-Werror,-Wignored-attributes]

jkotas commented on src/coreclr/vm/cgensys.h:
```diff
@@ -97,8 +97,14 @@ inline void GetSpecificCpuInfo(CORINFO_CPU * cpuInfo)
 
-extern "C" void __stdcall __cpuid(int cpuInfo[4], int function_id);
-extern "C" void __stdcall __cpuidex(int cpuInfo[4], int function_id, int subFunction_id);
+#ifdef TARGET_X86
+#define CALLCONV __cdecl
+#else
+#define CALLCONV __stdcall
+#endif
+
+extern "C" void CALLCONV __cpuid(int cpuInfo[4], int function_id);
+extern "C" void CALLCONV __cpuidex(int cpuInfo[4], int function_id, int subFunction_id);
```
```suggestion
extern "C" void __cpuid(int cpuInfo[4], int function_id);
extern "C" void __cpuidex(int cpuInfo[4], int function_id, int subFunction_id);
```
Does this work?

It is best to avoid explicit cdecl vs. stdcall calling conventions on non-Windows.

am11 replied:
It compiles on linux x86 and osx arm64. Lets see what CI thinks. :)

jkotas commented on src/coreclr/vm/cgensys.h:
```diff
@@ -97,8 +97,14 @@ inline void GetSpecificCpuInfo(CORINFO_CPU * cpuInfo)
+extern "C" void CALLCONV __cpuid(int cpuInfo[4], int function_id);
+extern "C" void CALLCONV __cpuidex(int cpuInfo[4], int function_id, int subFunction_id);
 #endif // TARGET_UNIX
 extern "C" DWORD __stdcall xmmYmmStateSupport();
```
You can delete explicit `__stdcall` here as well while you are on it. (It needs to be deleted in the implementation as well.)

am11 committed:
Remove calling convention modifiers from cpuid sig

> /runtime/src/coreclr/vm/cgensys.h:107:26: error: stdcall calling convention is not supported on builtin function [-Werror,-Wignored-attributes]

ayakael commented:
Passes on my alpine-x86 cross-cI: https://lab.ilot.io/ayakael/dotnet-stage0/-/jobs/1475

jkotas reviewed (approved):
Thanks!
